instrumentation/opentelemetry-instrumentation-logging/src/opentelemetry/instrumentation/logging/__init__.py

```diff
@@ -76,20 +76,29 @@ def instrumentation_dependencies(self) -> Collection[str]:
         return _instruments
 
     def _instrument(self, **kwargs):
-        service_name = ""
-        provider = kwargs.get("tracer_provider", None) or get_tracer_provider()
-        resource = provider.resource if provider else None
-        if resource:
-            service_name = resource.attributes.get("service.name")
 
+        provider = kwargs.get("tracer_provider", None) or get_tracer_provider()
         old_factory = logging.getLogRecordFactory()
         LoggingInstrumentor._old_factory = old_factory
 
+        service_name = None
+
         def record_factory(*args, **kwargs):
             record = old_factory(*args, **kwargs)
 
             record.otelSpanID = "0"
             record.otelTraceID = "0"
```

Comment:
Probably out of scope, what do you think about setting service name to "" here similar to span, traceid and defensively access resource on the below span instead?

Reply:
Not sure about accessing resource from span as span API does not define resource at all. ReadableSpan defines resource which should only be used by the SDK. I know somewhat similar argument applies to provider but for some reason I feel doing it with provider is a lesser evil. Happy to hear other ideas/counter-arguments but may be we shold open an issue and discuss there?

Reply:
Actually we should discuss it in the next SIG call. Not span vs provider but the fact that an instrumentation depends on an SDK feature which shouldn't be the case.

Reply:
> the fact that an instrumentation depends on an SDK feature which shouldn't be the case.

Isn't this an exception and I remember there was some discussion around if `instrumentation-logging` should actually be an instrumentation. I am not aware of any other instrumentation depending sdk feature.